plugins/inputs/system/memory.go

```diff
@@ -1,14 +1,21 @@
 package system
 
 import (
+	"bufio"
 	"fmt"
+	"os"
+	"strconv"
+	"strings"
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
+var linuxMeminfoPath = "/proc/meminfo" // so we can override it in tests
+
 type MemStats struct {
-	ps PS
+	ps           PS
+	LinuxMeminfo bool
```

Comment:
Ok, I follow now. We have to keep it around as the meminfo stuff only works on linux. gopsutil (aka "ps" here) works on multiple OSs. The other issue is it would be a breaking change.